src/Microsoft.Identity.Web.Certificate/DefaultCredentialsLoader.cs

```diff
@@ -81,11 +116,18 @@ public async Task LoadCredentialsIfNeededAsync(CredentialDescription credentialD
         }
 
         /// <inheritdoc/>
-        public async Task<CredentialDescription?> LoadFirstValidCredentialsAsync(IEnumerable<CredentialDescription> credentialDescriptions, CredentialSourceLoaderParameters? parameters = null)
+        /// Loads first valid credential which is not marked as Skipped. 
+        public async Task<CredentialDescription?> LoadFirstValidCredentialsAsync(
+            IEnumerable<CredentialDescription> credentialDescriptions, 
+            CredentialSourceLoaderParameters? parameters = null)
         {
             foreach (var credentialDescription in credentialDescriptions)
             {
                 await LoadCredentialsIfNeededAsync(credentialDescription, parameters);
+
+                // TODO:  Exceptions which occur in trying various credentials are logged and ignored.
```

Comment:
I didn't change here but this method looks broken because it doesn't handle exceptions. There is no coverage of this.